Sources/TuistCore/Cache/CacheDirectoriesProvider.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+public protocol CacheDirectoriesProviding: AnyObject {
+    /// Returns the cache directory
+    var cacheDirectory: AbsolutePath { get }
+
+    /// Returns the directory where the build artifacts are cached.
+    var buildCacheDirectory: AbsolutePath { get }
+
+    /// Returns the directory where hashes of modules that have been a part of successful test are cached
+    var testsCacheDirectory: AbsolutePath { get }
+
+    /// Returns the directory where the projects generated for automation tasks are generated to
+    var projectsCacheDirectory: AbsolutePath { get }
+
+    /// Returns the directory where the project description helper modules are cached.
+    var projectDescriptionHelpersCacheDirectory: AbsolutePath { get }
+
+    /// Returns the directory where the project description helper modules are cached.
+    var manifestCacheDirectory: AbsolutePath { get }
+}
+
+public final class CacheDirectoriesProvider: CacheDirectoriesProviding {
+    public let cacheDirectory: AbsolutePath
+    private static let defaultDirectory = AbsolutePath(URL(fileURLWithPath: NSHomeDirectory()).path).appending(component: ".tuist")
+
+    public init(config: Config?) throws {
+        if let cacheDirectory = config?.cache?.path {
```

Comment:
As I mentioned in another comment, Tuist internal caching and caching of the project modules are not the same thing and therefore we should make sure we differentiate them. I'd use `projectBinariesCachePath` to refer to `config.cache.path`, and `tuistCachePath`, to refer to the cache that Tuist uses internally. 
Using very explicit names will avoid confusion down the road.